src/index.js

```diff
@@ -1,35 +1,202 @@
+
+      todoArray[indx].completed = false;
+    }
+    localStorage.setItem('todoList', JSON.stringify(todoArray));
+  });
+};
+
+document.getElementById('todo-enter').addEventListener('click', () => {
+  const descriptionTodo = InputElementTodo.value;
+  InputElementTodo.value = '';
+  localStorage.setItem('input', InputElementTodo.value);
+  todoArray.push({
+    description: descriptionTodo,
     completed: false,
-    index: 1,
-  },
-];
+    index: todoArray.length,
+  });
+  localStorage.setItem('todoList', JSON.stringify(todoArray));
+  todoListRender(descriptionTodo, false, todoArray.length - 1);
+});
 
 todoArray.forEach((todo) => {
+  const todoListItemContainer = document.createElement('div');
+  todoListItemContainer.classList.add('todo-element-padding');
+  todoListItemContainer.classList.add('todo-item-container');
+
   const todoListCheck = document.createElement('input');
   todoListCheck.setAttribute('type', 'checkbox');
   todoListCheck.setAttribute('id', `todo-list-check-${todo.index}`);
-  todoListCheck.classList.add('todo-element-padding');
-  document.getElementById('todo-list-element').appendChild(todoListCheck);
+  todoListCheck.classList.add('test-check');
+  todoListItemContainer.appendChild(todoListCheck);
 
   const todoListElement = document.createElement('span');
-  todoListElement.classList.add('todo-element-padding');
+  todoListElement.classList.add('ellipsis');
   todoListElement.innerText = todo.description;
 
-  document.getElementById('todo-list-element').appendChild(todoListElement);
+  todoListItemContainer.appendChild(todoListElement);
+
   const threeDotIcon = document.createElement('span');
   threeDotIcon.innerHTML = '&#8942;';
-  document.getElementById('todo-list-element').appendChild(threeDotIcon);
+  todoListItemContainer.appendChild(threeDotIcon);
+
+  document
+    .getElementById('todo-list-element')
+    .appendChild(todoListItemContainer);
 
   document
     .getElementById('todo-list-element')
     .appendChild(document.createElement('hr'));
+
+  todoListCheck.addEventListener('click', () => {
+    if (todoListCheck.checked === true) {
+      todoListCheck.nextElementSibling.style.textDecoration = 'line-through';
+      todoListCheck.nextElementSibling.style.opacity = '0.5';
+      const indexOfTodoElement = todoListCheck.id.split('').pop();
+      todoArray.forEach((todo) => {
+        if (todo.index === Number(indexOfTodoElement)) {
+          todo.completed = true;
+        }
+      });
+      localStorage.setItem('todoList', JSON.stringify(todoArray));
+    } else {
+      todoListCheck.nextElementSibling.style.textDecoration = 'none';
+      todoListCheck.nextElementSibling.style.opacity = '1';
+      const indexOfTodoElement = todoListCheck.id.split('').pop();
+      todoArray.forEach((todo) => {
+        if (todo.index === Number(indexOfTodoElement)) {
+          todo.completed = false;
+        }
+      });
+      localStorage.setItem('todoList', JSON.stringify(todoArray));
```

Comment:
- [ ] As required for the project, kindly ensure that by default, new tasks have the property `index` set to the value of the new array length (i.e. if a `1st` task is added to the list, the index of that task should equal to `1`, if a `4th` task is added to the list, its index property should have a value of `4`,...).